google/cloud/spanner_v1/_opentelemetry_tracing.py

```diff
@@ -15,41 +15,89 @@
 
     HAS_OPENTELEMETRY_INSTALLED = True
+    DB_SYSTEM = SpanAttributes.DB_SYSTEM
+    DB_NAME = SpanAttributes.DB_NAME
+    DB_CONNECTION_STRING = SpanAttributes.DB_CONNECTION_STRING
+    NET_HOST_NAME = SpanAttributes.NET_HOST_NAME
+    DB_STATEMENT = SpanAttributes.DB_STATEMENT
 except ImportError:
     HAS_OPENTELEMETRY_INSTALLED = False
 
 
+EXTENDED_TRACING_ENABLED = os.environ.get('SPANNER_ENABLE_EXTENDED_TRACING', '') == 'true'
+
+SPAN_NAMESPACE = 'cloud.google.com/python/spanner'
+TRACER_NAME = SPAN_NAMESPACE
+
+def get_tracer(tracer_provider=None):
+    """
+    get_tracer is a utility to unify and simplify retrieval of the tracer, without
+    leaking implementation details given that retrieving a tracer requires providing
+    the full qualified library name and version.
+    When the tracer_provider is set, it'll retrieve the tracer from it, otherwise
+    it'll fall back to the global tracer provider and use this library's specific semantics.
+    """
+    if tracer_provider:
+        return tracerProvider.get_tracer(TRACER_NAME, TRACER_VERSION)
+    else:
+        return trace.get_tracer(TRACER_NAME, TRACER_VERSION)
+
 @contextmanager
-def trace_call(name, session, extra_attributes=None):
+def trace_call(name, session, extra_attributes=None, observability_options=None):
     if not HAS_OPENTELEMETRY_INSTALLED or not session:
         # Empty context manager. Users will have to check if the generated value is None or a span
         yield None
         return
 
-    tracer = trace.get_tracer(__name__)
+    tracer = None
+    if observability_options:
+        tracerProvider = observability_options.get('tracer_provider', None)
+        if tracerProvider:
+            tracer = get_tracer(tracerProvider)
+
+    if tracer is None: # Acquire the global tracer if none was provided.
+        tracer = get_tracer()
+
+    # It is imperative that we properly record the Cloud Spanner
+    # endpoint tracks whether we are connecting to the emulator
+    # or to production.
+    spanner_endpoint = os.getenv("SPANNER_EMULATOR_HOST")
+    if not spanner_endpoint:
+        spanner_endpoint = SpannerClient.DEFAULT_ENDPOINT
 
     # Set base attributes that we know for every trace created
     attributes = {
-        "db.type": "spanner",
-        "db.url": SpannerClient.DEFAULT_ENDPOINT,
-        "db.instance": session._database.name,
-        "net.host.name": SpannerClient.DEFAULT_ENDPOINT,
+        DB_SYSTEM: "spanner",
+        DB_CONNECTION_STRING: spanner_endpoint
+        DB_NAME: session._database.name,
+        NET_HOST_NAME: spanner_endpoint,
     }
 
     if extra_attributes:
         attributes.update(extra_attributes)
 
+    extended_tracing = EXTENDED_TRACING_ENABLED or (
+                        observability_options and
+                        observability_options.get('enable_extended_tracing', False))
+
+    if not extended_tracing:
+        attributes.pop(DB_STATEMENT, None)
+
+    span_name = SPAN_NAMESPACE + '/' + name
```

Comment:
What is the trace span output with this change?
I assume the span name will change from `CloudSpanner.CreateSession` to `cloud.google.com/python/spanner/CloudSpanner.CreateSession`. If yes this change is not needed, span name can be as it is and need not be prefixed with namespace.